File: Sources/structly/04_linked_lists/SumList.swift
//
//  SumList.swift
//  MyCLI
//
//  Created by new on 3/16/26.
//

func sumList(_ head: Node<Int>?) -> Int {
    guard let node = head else { return 0 }
    return node.val + sumList(node.next)
}

func sumList1(_ head: Node<Int>?) -> Int {
    var current = head
    var res = 0
    while let node = current {
        res += node.val
        current = node.next
    }
    return res
}

File: Sources/structyC/04_linked_list/sum_list.cpp
//
//  sum_list.cpp
//  MyCLI
//
//  Created by new on 3/16/26.
//

class Node {
  public:
    int val;
    Node *next;

    Node(int initialVal) {
        val = initialVal;
        next = nullptr;
    }
};

int sumList(Node *head) {
    if (head == nullptr) {
        return 0;
    }
    return head->val + sumList(head->next);
}

int sumList1(Node *head) {
    int sum = 0;
    Node *node = head;
    while (node != nullptr) {
        sum += node->val;
        node = node->next;
    }
    return sum;
}
